sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -37,6 +37,20 @@ import org.apache.spark.util.{MutableURLClassLoader, Utils}
  */
 private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
 
+  // System preserved database should not exists in metastore. However it's hard to guarantee it
+  // for every session, because case-sensitivity differs. Here we always lowercase it to make our
+  // life easier.
+  val globalTempDB = sparkContext.conf.get(GLOBAL_TEMP_DATABASE).toLowerCase
```

Comment:
yea, we do: https://github.com/apache/spark/pull/14897/files#diff-42e78d37f5dcb2a1576f83b53bbf4b55R46